content/articles/understanding-cross-site-scripting-attacks/index.md

```diff
@@ -0,0 +1,123 @@
+
+The next step is to locate an input field where we can paste our script. It could be a username, a search box, a text area, or any other type of input field. Our vulnerable website has a search box, perfect! Now, let's give it some simple HTML code and see what happens. Type in the following HTML code in the search box and hit the go button:
+
+```html
+<u>test</u>
+<!-- the HTML u tag or underline tag is used to give a word an underline in HTML -->
+```
+
+![html](/understanding-cross-site-scripting-atacks/u.png)
+
+From the screenshot above we can see that the word `test` is underlined. Perfect! we are getting a reflection on the page. By viewing the application's source code, we can also see how it's reflected.
+
+![view-page source](/understanding-cross-site-scripting-atacks/test1.png)
+
+Now let's look for XSS. The most simple XSS payload is the script tag. A script tag can either reference or embed JavaScript code. One of the most common XSS payloads is the use of the javascript alert function in a script tag. Let's try this by entering the following payload in our web application's search box:
+
+```javascript
+<script>alert(1)</script>
+```
+
+![alert](/understanding-cross-site-scripting-atacks/alert.png)
+
+We notice that as soon as we hit the go button after entering our XSS payload, we get an alert box with `1` displayed, which means our payload worked successfully. We just used the [alert](https://www.w3schools.com/jsref/met_win_alert.asp) function within JavaScript to display 1.
+
+### Types of XSS
+
+#### Reflected XSS
+
+The victim's request is used to inject the payload. The victim must either click on a malicious link or visit a property controlled by the attacker.
+
+#### Stored XSS
+
+Also known as `Persistent XSS`, the payload is saved on the server and can be triggered by a victim without requiring any user interaction outside of the application. This type of XSS happens when the server saves your supplied input somewhere into the server (i.e. a database or cache server).
+
+##### Stored XSS Example
+
+Let's go back to our vulnerable web application. We click on the `Your profile` tab, which presents us with a login page
+
+![profile-login](/understanding-cross-site-scripting-atacks/profile-login.png)
+
+Now let's log in with these default credentials: 
+
+```
+username: test
+password: test
+```
+
+![logged in](/understanding-cross-site-scripting-atacks/in.png)
+
+Now that we are logged in, let's inject our payload in the name field and click on the update button to update our profile:
+
+```javascript
+<script>alert(1)</script>
+```
+
+![stored](/understanding-cross-site-scripting-atacks/stored.png)
+
+As we can see, we got an alert box. This occurred because the payload was saved in the name field, and the alert box was triggered every time the page was loaded.
+
+#### DOM XSS
+
+The vulnerability lies in the client-side code rather than the server-side code. Injections are still usually requested by the victim.
```

Comment:
Like what I said in the section about reflected XSS, this also sounds a little vague. I would try to give a more detailed explanation here if possible.

Reply:
Fixed, kindly go through. Thanks